lib/source-destination/source-destination.ts

```diff
@@ -77,8 +42,41 @@ import {
 } from './progress';
 import { SourceSource } from './source-source';
 
+class HashStream extends TransformStream {
+	private _outEnc: string;
+	private _hash: XXHash3;
+	private _readableState: any;
+	constructor(seed: number, outEnc: string | Buffer) {
+		super();
+		if (outEnc && typeof outEnc !== 'string' && !Buffer.isBuffer(outEnc)) {
+			outEnc = 'buffer';
+		}
+
+		this._outEnc = outEnc as string;
+
+		this._hash = new XXHash3(seed);
+
+		// Hack needed to support `readableObjectMode` in node v0.10 and to set
+		// `highWaterMark` only for Readable side
+		const rs = this._readableState;
+		rs.objectMode = true;
+		rs.highWaterMark = 2;
+	}
+
+	_transform(chunk: Buffer, _encoding: string, callback: () => void) {
+		this._hash.update(chunk);
+		callback();
+	}
+	_flush(callback: () => void) {
+		if (this._outEnc !== undefined) {
+			this.push(this._hash.digest(this._outEnc));
+		} else {
+			this.push(this._hash.digest());
+		}
+		callback();
```

Comment:
Isn't this the same as `this.push(this._hash.digest(this._outEnc))` ?

Reply:
Yes, you're right